arguebuf/load/_load_aif.py

```diff
@@ -16,17 +17,230 @@
+                                            # Find and update the hanging_node and argument_node in updated_hanging_nodes
+                                            for node in updated_hanging_nodes:
+                                                if node.node_id == hanging_node.node_id:
+                                                    node.outgoing_edges.append(new_edge_1)
+                                                if node.node_id == argument_node.node_id:
+                                                    node.incoming_edges.append(new_edge_2)
+    # check for loops:
+    remove_loops_in_hanging_nodes(updated_hanging_nodes, obj, nodes)
+    return obj
+
+
+def create_and_process_nodes(obj):
+    # ================================
+    # create and process simple nodes that will contain all their incoming/outgoing edges
+    nodes = {}
+    for aif_node in obj["nodes"]:
+        node_id = aif_node['nodeID']
+        node_text = aif_node.get('text', '')
+        node_type = aif_node.get('type', '')
+        node_timestamp = aif_node.get('timestamp', '')
+        nodes[node_id] = NewNode(node_id, node_text, node_type, node_timestamp)
+
+    # process edges and link them to nodes
+    for aif_edge in obj["edges"]:
+        edge_id = aif_edge['edgeID']
+        from_id = aif_edge['fromID']
+        to_id = aif_edge['toID']
+        edge_data = {'edgeID': edge_id, 'fromID': from_id, 'toID': to_id}
+
+        if from_id in nodes:
+            nodes[from_id].add_outgoing_edge(edge_data)
+        if to_id in nodes:
+            nodes[to_id].add_incoming_edge(edge_data)
+    return nodes
+
+
+def find_hanging_nodes(nodes):
+    # find nodes with no outgoing edges and exactly one incoming edge, and not of type "L"
+    # These are all the Nodes that have been left hanging in OVA representation and have no edges in the final graph
+    # with the original code, lets name them "hanging_nodes"
+    hanging_nodes = [node for node in nodes.values() if
+                    len(node.incoming_edges) == 1 and
+                    not node.outgoing_edges and
+                    node.node_type != "L"]
+    return hanging_nodes
+
+
+def process_hanging_nodes(obj):
+    nodes = create_and_process_nodes(obj)
+    hanging_nodes = find_hanging_nodes(nodes)
+    updated_hanging_nodes = hanging_nodes.copy()
+    obj = process_each_hanging_node(hanging_nodes, obj, nodes, updated_hanging_nodes)
+    return obj
+
+
 def load_aif(
-    obj: aif.Graph,
-    name: t.Optional[str] = None,
-    config: Config = DefaultConfig,
+        obj: aif.Graph,
+        name: t.Optional[str] = None,
+        config: Config = DefaultConfig,
 ) -> Graph:
     """Generate Graph structure from AIF argument graph file
     (reference: http://www.wi2.uni-trier.de/shared/publications/2019_LenzOllingerSahitajBergmann_ICCBR.pdf)
 
     """
     g = config.GraphClass(name)
 
+    # Process hanging nodes
+    obj = process_hanging_nodes(obj)
```

Comment:
Maybe we should make this configurable by adding a parameter `process_hanging_nodes: bool = True` to the function `load_aif`?